src/modules/assignment/assignment.service.ts

```diff
@@ -39,28 +65,68 @@ export class AssignmentService {
   async addAssignment(
     CreateAssignmentDTO: CreateAssignmentDTO,
   ): Promise<Assignment> {
-    const newAssignment = await new this.AssignmentModel(CreateAssignmentDTO);
-    return newAssignment.save();
+    try {
+      const newAssignment = await new this.AssignmentModel(CreateAssignmentDTO);
+      await newAssignment.save();
+      if (newAssignment) {
+        return newAssignment;
+      }
+      throw new NotFoundException('doubt not found!');
+    } catch (e) {
+      throw new HttpException(
+        {
+          status: HttpStatus.INTERNAL_SERVER_ERROR,
+          error: `${e}`,
+        },
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
   }
 
   // Edit Assignment details
   async updateAssignment(
     AssignmentID,
     updateAssignmentDTO: UpdateAssignmentDTO,
   ): Promise<Assignment> {
-    const updatedAssignment = await this.AssignmentModel.findByIdAndUpdate(
-      AssignmentID,
-      updateAssignmentDTO,
-      { new: true },
-    );
-    return updatedAssignment;
+    try {
+      const updatedAssignment = await this.AssignmentModel.findByIdAndUpdate(
+        AssignmentID,
+        updateAssignmentDTO,
+        { new: true },
+      );
+      if (updatedAssignment) {
+        return updatedAssignment;
+      }
+    } catch (e) {
+      throw new HttpException(
+        {
+          status: HttpStatus.INTERNAL_SERVER_ERROR,
+          error: `${e}`,
+        },
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+    throw new NotFoundException('doubt not found!');
```

Comment:
Already told you to replace the doubts with assignments :(

Reply:
this is a different branch
PE#90 has the changes made